### PR TITLE
feat: get filters for address and estate

### PR DESCRIPTION
**Pull request by kauffinger**

### CODE REVIEW

File: tests/Stubs/GetFiltersResponse.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            'type' => 'filter',
+            'elements' => [
+                'scope' => fake()->randomElement(['estate', 'address']),
+                'name' => fake()->words(3, true),
+                'userId' => fake()->optional()->randomNumber(),
+                'groupId' => fake()->randomNumber(),
+            ],
+        ]);
+
+        return array_merge_recursive([
```
Comment (Katalam) [inline tests/Stubs/GetFiltersResponse.php:30]:
```suggestion
        return array_replace_recursive([
```

File: src/Query/FilterBuilder.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        return $this;
+    }
+
+    public function address(): static
+    {
+        $this->module = 'address';
+
+        return $this;
+    }
```
Comment (Katalam) [inline src/Query/FilterBuilder.php:81]: Could be a trait, we have it already somewhere else

**Review by Katalam — CHANGES_REQUESTED** [review]
No description will be closed when marked as ready.

Changes as requested

File: src/Query/FilterBuilder.php (hunk 2 above)
Comment (kauffinger) [reply src/Query/FilterBuilder.php:81]: I don't think so, field builder has https://github.com/innobraingmbh/laravel-onoffice-adapter/blob/1771811ab083e70d836f7f12d0bfe2956317d0b1/src/Query/FieldBuilder.php#L17C3-L17C32
which is a modules array.

ActivityBuilder has https://github.com/innobraingmbh/laravel-onoffice-adapter/blob/1771811ab083e70d836f7f12d0bfe2956317d0b1/src/Query/ActivityBuilder.php#L136C1-L148C6
which also does something different (and will need to be changed as activities can be linked to addressIds and an estateId at the same time)

**Review by Katalam — APPROVED** [review]

**Comment by Katalam** [comment]
Should we implement a QueryException? When we fire a request without a specific module, we will get an exception that we accessed uninitialized variables. Maybe we should check for that a throw a specific exception?

**Pull request merged**